Source/Core/Common/QoSSession.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <enet/enet.h>
+
+namespace Common
+{
+class QoSSession
+{
+public:
+  // 1 0 1 1 1 0 0 0
+  // DSCP      ECN
+  static constexpr int ef_tos = 0b10111000;
+
+  QoSSession() = default;
+  QoSSession(ENetPeer* peer, int tos_val = ef_tos);
+
+  ~QoSSession();
+
+  QoSSession& operator=(const QoSSession&) = delete;
+  QoSSession(const QoSSession&) = delete;
+
+  QoSSession& operator=(QoSSession&& session)
+  {
+    if (this != &session)
+    {
+#if defined(_WIN32)
+      m_qos_handle = session.m_qos_handle;
+      m_qos_flow_id = session.m_qos_flow_id;
+
+      session.m_qos_handle = nullptr;
+      session.m_qos_flow_id = 0;
+#endif
+      m_peer = session.m_peer;
+      m_success = session.m_success;
+
+      session.m_peer = nullptr;
+      session.m_success = false;
+    }
+
+    return *this;
+  }
+
+  QoSSession(QoSSession&& session)
+      :
+#if defined(_WIN32)
+        m_qos_handle(session.m_qos_handle), m_qos_flow_id(session.m_qos_flow_id),
+#endif
+        m_peer(session.m_peer), m_success(session.m_success)
+  {
+    if (this != &session)
```

Comment:
You don't need to check for self assignment in a constructor (if a new object is being constructed, then `this` can never be equal to `session),, which is why I only mentioned the operators, but not constructors in my comment.

Even then, if it *was* required, then there would be a bug, since the initializer list would be copying everything over regardless of the check :p

Reply:
Oops, that was a copy+paste error lol